earthdata_download/download.py

```diff
@@ -122,12 +126,16 @@ def data_url_from_entry(entry, **kwargs):
 
 def _download_file_https(url, target, username, password):
     target_temp = target + '.incomplete'
-    with EarthdataSession(username=username, password=password) as session:
-        with session.get(url, stream=True) as response:
-            response.raise_for_status()
-            response.raw.decode_content = True
-            with open(target_temp, "wb") as target_file:
-                shutil.copyfileobj(response.raw, target_file)
+    with EarthdataSession(username=username, password=password) as session: 
+        with session.get(url=url, stream=True) as r:        
+            r.raise_for_status()
+            r.raw.decode_content = True
+            with open(target_temp, "wb") as t:
+                if session.DefaultUserAgent:
+                    t.write(r.content)
+                else:
+                    shutil.copyfileobj(r.raw, target_temp) 
```

Comment:
The `shutil.copy` should not be inside the `with open(target_temp, ...)` context.

Reply:
I think it is fine now. 

Realized, I must have tested this on an older version of the code.  I am puzzled how my test-download could work. The `if else` was in the wrong order:
```python
if session.DefaultUserAgent:
    t.write(r.content)
else:
    shutil.copyfileobj(r.raw, target_temp) 
```

And I found, that `user_agent` was never transfered through the workflow.